src/store/CurrentAccount/actions.ts

```diff
@@ -0,0 +1,41 @@
+import {CurrentAccountInterface} from "@/interfaces";
+import {SessionStorageKeys} from "sbc-common-components/src/util/constants";
+import {sleep} from "@/utils";
+import currentAccount from "@/store/CurrentAccount/index";
+
+export default {
+  async loadCurrentAccount (context): Promise<any> {
+    return new Promise((resolve, reject) => {
+      context.dispatch('waitForCurrentAccount')
+        .then((currentAccount) => {
+          const accountInfo: CurrentAccountInterface = {
+            accountType: currentAccount.accountType,
+            id: currentAccount.id,
+            label: currentAccount.label,
+            type: currentAccount.type
+          }
+          context.commit('setAccountInformation', accountInfo)
+          resolve()
+        })
+        .catch(() => {
+          reject()
+        })
+    })
+  },
+
+  async waitForCurrentAccount (): Promise<any> {
+    return await new Promise((resolve, reject) => {
+      let account = null
+      for (let i = 0; i < 50; i++) {
+        const currentAccount = sessionStorage.getItem(SessionStorageKeys.CurrentAccount)
+        account = JSON.parse(currentAccount)
+        if (account) break
+          sleep(100)
+      }
+      if (account) {
+        return resolve(account)
+      }
+      return reject()
+    })
```

Comment:
I _think_ resolving or rejecting a returning promise is equivalent to returning when done or throwing an error. It might make for simpler code?

Reply:
(Async methods automatically have a return type of `<Promise<any>>`.

Reply:
Hmm, I don't think that's true.  The beauty of using promises is that caller can handle a promise-resolve differently from a promise-reject.  For example, I could imagine wanting to do something like this in the future.
```
await this.fetchSomeResourceFromApi()
  .catch((error) => {
    commit('addApiErrorMessage', error)
  })
```

Reply:
A called can use `.then()` for a resolved promise and `.catch()` for a rejected promise.

Or just await the async method if you don't care about errors.